Mathlib/MeasureTheory/Constructions/BorelSpace/Basic.lean

```diff
@@ -182,7 +182,7 @@ instance (priority := 100) BorelSpace.opensMeasurable {α : Type*} [TopologicalS
   ⟨ge_of_eq <| BorelSpace.measurable_eq⟩
 
 instance Subtype.borelSpace {α : Type*} [TopologicalSpace α] [MeasurableSpace α]
-    [hα : BorelSpace α] (s : Set α) : BorelSpace s :=
+    [hα : BorelSpace α] (p : α → Prop) : BorelSpace (Subtype p) :=
```

Comment:
For other reviewers: the change in this file is a syntactic generalisation to this instance and a related one below, suggested by sgouezel, which allows removing the two local duplicated instances in Disintegration.lean, and also let Lean find the BorelSpace instance in the new `hausdorffMeasure_orthogonalProjection_le` lemma. These changes aren't that disjoint from this PR: they're needed for it but motivated by it, so in my opinion it's fine to keep this as one PR rather than split it.